contracts/truefi2/CollateralVault.sol

```diff
@@ -43,13 +51,13 @@ contract CollateralVault is ICollateralVault, UpgradeableClaimable {
     }
 
     function stake(uint256 amount) external {
-        // require(borrowingMutex.isUnlocked(borrower) || borrowingMutex.locker(borrower) == LOCA)
-        // if borrowingMutex.locker(borrower) == LOCA:
-        //   poke LOCA for borrower to update limit + rate
-        // safe transfer amount from borrower
-        require(amount == 0); // silence lint
-        stakedToken = IERC20WithDecimals(address(0)); // silence build warning
-        revert("Unimplemented!");
+        require(
+            borrowingMutex.isUnlocked(msg.sender) || borrowingMutex.locker(msg.sender) == address(lineOfCreditAgency),
+            "CollateralVault: Borrower can only stake when they're unlocked or have a line of credit"
+        );
+        stakedToken.safeTransferFrom(msg.sender, address(this), amount);
+        stakedAmount[msg.sender] = stakedAmount[msg.sender].add(amount);
```

Comment:
I just remembered we should swap these two lines for checks-effects-interactions.

Technically the event emission should also go before the safeTransferFrom, but for consistency with the rest of the repo we may as well leave it at the end where it is.